src/main/java/com/powsybl/openloadflow/network/impl/LfShuntImpl.java

```diff
@@ -60,6 +67,13 @@ public Evaluable getQ() {
 
     @Override
     public void updateState() {
-        shuntCompensator.getTerminal().setQ(q.eval() * PerUnit.SB);
+        double qCalc = q.eval();
+        double bSum = shuntCompensators.stream()
+                .mapToDouble(ShuntCompensator::getB)
+                .sum();
```

Comment:
Be careful here because even if we have one or several shunts connected to a bus, the sum could be equal to zero.